pkg/util/fdlimit/fdlimit_unix.go

```diff
@@ -17,6 +17,9 @@ func SetMaxFDs(max uint64) (uint64, error) {
 	if max > rLimit.Max {
 		max = rLimit.Max
 	}
+	if rLimit.Cur > max {
+		return rLimit.Cur, nil
+	}
```

Comment:
I think that isn't correct. In this case caller can't decrease max file descriptors amount.